guillotina_rediscache/lru_sized.py

```diff
@@ -0,0 +1,99 @@
+
+from lru import LRU
+from sys import getsizeof
+
+
+class ObjectToLarge(Exception):
+    pass
+
+
+class TooManyKeys(Exception):
+    pass
+
+
+class LRUSized:
+    """
+    A memory sized implementation of a LRUDict
+    Instead of storing the values, for each
+    value we store a tuple of the value and it's size,
+    that allows us to track the current size, in an
+    easy way.
+    """
+    def __init__(self, size, max_bytes):
+
+        def evicted(key, value):
+            self._size -= value[1]
+
+        self._lru = LRU(size, callback=evicted)
+        self.max_bytes = max_bytes
+        self._size = 0
+
+    def __contains__(self, key):
+        return key in self._lru
+
+    def __getitem__(self, item):
+        return self._lru[item][0]
+
+    def __setitem__(self, key, val):
+        size = getsizeof(val)
```

Comment:
I think we're storing bytes right? Should we use `len()` instead? I know there are some funny things about getsizeof that are counterintuitive.

Reply:
Yes, on the end they are bytes, but I'm not sure if I can access the bytesize (without) pickling, that is something that happens later ( https://github.com/onna/guillotina_rediscache/blob/master/guillotina_rediscache/cache_strategy.py#L71). Anyway I think we don't want to incur on the cost of pickling again to get the  object size... Let me think, perhaps I'm on a wrong approach.. And memory management should be implemented on the RedisCacheClass. (That could allow us to release memory on async approach), without the cost of double pickling..